Make `exec` member of add-on manifests optional for extensions again

In the TypeScript re-write for 1.1 a change was made that made the exec member of add-on manifests mandatory.

This means it's no longer possible to create an extension add-on which is just CSS and JavaScript content with no back end executable.

We should consider making exec optional again to allow extension add-ons (e.g the square theme).

Note to self: If I revert the added line in the object literal which defines `exec: ''` I get an error:

`src/addon-utils.ts:125:58 - error TS2339: Property 'exec' does not exist on type '{ primary_type: string; }'.`

If I comment out line 125:

`manifestTemplate.gateway_specific_settings.webthings.exec = '';`

then the square theme add-on will install successfully, but presumably that will break validation for add-ons which aren't extensions.

--- a/src/addon-utils.ts
+++ b/src/addon-utils.ts
@@ -88,7 +88,6 @@ function validateManifestJson(manifest: Record<string, unknown>): string | null 
     gateway_specific_settings: {
       webthings: {
         primary_type: '',
-        exec: '',
       },
     },
     homepage_url: '',
@@ -123,7 +122,9 @@ function validateManifestJson(manifest: Record<string, unknown>): string | null 
   ) {
     // If we're not using in-process plugins, and this is not an extension,
     // then we also need the exec keyword to exist.
-    manifestTemplate.gateway_specific_settings.webthings.exec = '';
+    (<Record<string, unknown>>(
+      (<Record<string, unknown>>manifest.gateway_specific_settings).webthings
+    )).exec = '';
   }
 
   return validateObject('', manifest, manifestTemplate);

